tests/testthat/test-shinytest2-landing_popup.R

```diff
@@ -0,0 +1,184 @@
+testthat::test_that("e2e: teal app with landing_popup_module initializes with no errors", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(
+        title = "Welcome",
+        content = tags$b("A welcome message!", style = "color: red;")
+      ),
+      example_module()
+    )
+  )
+  app$view()
+
+  app$wait_for_idle(timeout = default_idle_timeout)
+  app$expect_no_shiny_error()
+  app$stop()
+})
+
+test_that("e2e: app with default landing_popup_module creates modal containing a button", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_equal(
+    app$get_html(".btn-default") %>%
+      grep("Accept", x = ., value = TRUE) %>%
+      rvest::read_html() %>%
+      rvest::html_text(),
+    "Accept"
+  )
+
+  app$stop()
+})
+
+test_that("e2e: when default landing_popup_module is closed, it shows the underlying teal app", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  app$click(selector = ".btn-default:nth-child(1)")
+
+  # QUESTION: Is there a better way of checking the name of current module?
+  expect_equal(
```

Comment:
Consider making sure the modal is gone from DOM. As the test passes even if button is not clicked.

```r
testthat::expect_null(app$get_html("#shiny-modal-wrapper"))
```

Reply:
Is there a real need to know which module is active?

The Teal app initializes with the first module as active. Then the navigation made in the test logic is the only way to change the module so it should be quite clear which is the active module because the test code should have this information.

Reply:
Yeah, just wanted to double check if the modal is closed.

Reply:
d7a99d55837db5d07eabdd34ca33e28ffaa00edf